webui/react/src/services/stream/keyCache.ts

```diff
@@ -0,0 +1,82 @@
+export function decode_keys(keys: string): Array<number>{
+    const retval: Array<number> = [];
+    if (keys.length === 0) {
+        return retval;
+    }
+    for (const key of keys.split(',')) {
+        if (key.includes('-')) {
+            const [start, end] = key.split('-');
+            for (let i = Number(start); i <= Number(end); i++) {
+                retval.push(i);
+            }
+        } else {
+            retval.push(Number(key));
+        }
+    }
+    return retval;
+}
+
+export function encode_keys(set: Set<number>): string {
+    if (!set || set.size === 0) return '';
+    const out: Array<string> = [];
+    const keys = Array.from(set).sort((l, r) => l - r).values();
```

Comment:
doesn't `Array.sort()` do the right thing by default?  I think you could skip the custom comparator function here, if that's true

Reply:
I haven't looked too deep into this, but I was running into the issue that [2,10,63] being sorted into [10,2,63] with the default sort

Reply:
oh, that would be a string sort, since `1` of `10` comes before `2`.

Sure enough, `[63, 10, 2].sort()` returns `[2, 10, 63]` exactly like you say in node.

TIL.